pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -230,6 +232,28 @@ func (cl *Client) Update(config config.Config) (string, error) {
 	return meta.GetResourceVersion(), nil
 }
 
+// PatchFunc provides the cached config as a base for modification. The diff between the input value and the modified
+// return will be used to apply a patch in Kubrenetes.
+type PatchFunc func(cfg config.Config) config.Config
+
+// Patch applies only the modifications made in the PatchFunc rather than doing a full replace. Useful to avoid
+// read-modify-write conflicts when there are many concurrent-writers to the same resource.
+func (cl *Client) Patch(typ config.GroupVersionKind, name, namespace string, patchFn PatchFunc) (string, error) {
```

Comment:
This isn't exposed on the `ConfigStore` interface because it's pretty Kubernetes specific. The only other "writable" config store is the in-memory one, and I doubt we want to implement patching logic there.